codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt

```diff
@@ -144,6 +150,27 @@ class PresignableUrlIntegration(private val presignedOperations: Map<String, Set
         }
     }
 
+    private fun renderPresignURLAPIInServiceClient(
+        writer: SwiftWriter,
+        clientName: String,
+        op: OperationShape,
+        inputType: String
+    ) {
+        writer.apply {
+            openBlock("extension $clientName {", "}") {
+                val params = listOf("input: $inputType", "expiration: Foundation.TimeInterval")
+                val returnType = "ClientRuntime.URL"
+                openBlock("public func presignURL${op.toUpperCamelCase()}(${params.joinToString()}) async throws -> $returnType {", "}") {
+                    write("let presignedURL = try await input.presignURL(config: config, expiration: expiration)")
+                    openBlock("guard let presignedURL else {", "}") {
+                        write("throw ClientError.unknownError(\"Returned URL from input.presignURL() was nil.\")")
```

Comment:
To help make it clear where this error originated, what do you think of including the operation name in the error message?

Reply:
That would be great, will add that in